pages/en/lb4/Controllers.md

```diff
@@ -156,3 +156,52 @@ class HelloController {
 - A `Repository` is LoopBack's database abstraction. See [Repositories](Repositories.html) for more.
 - `@get('/messages')` creates the `Route` for the Operation using `app.route()`.
 - `@param.query.number` adds a `number` param with a source of `query`.
+
+## Handling Errors in Controllers
+
+In order to specify errors for controller methods to throw, the class `HttpErrors` is used. `HttpErrors` can be found in the `@loopback/rest` package.
+
+The example below shows how `HttpErrors` can be used inside a controller method and how it can be tested.
+
+```js
+// the test
+import {HelloController} from 'path.to.controller'
+import {expect} from '@loopback/testlab'
+// ...
+
+// ...
+describe('Hello Controller', () => {
+  // ...
+  it('returns 412 Precondition Failed for non-positive limit', () => {
+    const controller = new HelloController();
+    try {
+      controller.list(0);
+      throw new Error('should have thrown an error');
```

Comment:
why are we throwing a new error here? shouldn't 
```
try { 
controller.list(0);
} catch (err) {
...
}
```
be enough?

Reply:
Good catch! Sorry I missed that. Actually, it's even worse because the catch block will catch it instead.

Reply:
what happens if controller.list(0) doesn't throw error as expected? Shouldn't expects() in the catch blocks not run and cause the test to pass when it shouldn't?

Reply:
Move `expect` statements to be outside of the try/catch block and you can assign `err` to another variable in the catch block so it's accessible outside. 

```
try {
  controller.list(0)
} catch (err) {
  goterr = err
}
expect(goterr).to.exist();
// ... 
```

Reply:
I am personally fine with this approach. If we switch the order of the two `expect()` lines below and check the message first, then we will get a nice test failure message too - something along the lines "expected /precondition failed/" but got "should have thrown an error".

Reply:
❗️ I am pretty certain this test will be confusing to many other readers too. Regardless of the  approach we decide to follow, there should be code comment(s) explaining why and how this test works, why it's important to be prepared for the case when the controller method does not throw any error, and how we handled that.